platform/measurement_utils.cpp

```diff
@@ -189,4 +192,97 @@ string FormatSpeed(double metersPerSecond)
   return res;
 }
 
+bool OSMDistanceToMeters(string const & osmRawValue, double & outMeters)
+{
+  char * stop;
+  char const * s = osmRawValue.c_str();
+  outMeters = strtod(s, &stop);
+
+  // Not a number, was not parsed at all.
+  if (s == stop)
+    return false;
+
+  if (!isfinite(outMeters))
+    return false;
+
+  switch (*stop)
+  {
+  // Default units - meters.
+  case 0: return true;
+
+  // Feet and probably inches.
+  case '\'':
```

Comment:
Потому что такой код очень простой и читаемый, и легко расширяемый при желании.
И он делает ровно то, что ты предлагаешь.

> On 18.3.2016, at 15.18, ygorshenin <notifications@github.com> wrote:
> 
> In platform/measurement_utils.cpp:
> 
> > +  outMeters = strtod(s, &stop);
> > +
> > +  // Not a number, was not parsed at all.
> > +  if (s == stop)
> > +    return false;
> > +
> > +  if (!isfinite(outMeters))
> > +    return false;
> > +
> > +  switch (*stop)
> > +  {
> > +  // Default units - meters.
> > +  case 0: return true;
> > +
> > +  // Feet and probably inches.
> > +  case '\'':
> В целом опять получается та же лапша - почему ты не хочешь завести пару функций:
> 
> size_t EatNumber(string const & s, size_t pos, int & n);
> size_t EatMetric(string const & s, size_t pos, string & metric);
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>